Compiler/src/Compiler.cpp

```diff
@@ -1477,6 +1480,70 @@ struct Compiler
         }
     }
 
+    void compileExprInterpString(AstExprInterpString* expr, uint8_t target, bool targetTemp)
+    {
+        std::string formatString;
```

Comment:
Needs formatString.reserve(total number of characters), plus accounting for escaped % when computing that, to make sure that there's only one temporary allocation here.

Reply:
From the very beginning? What is the most efficient way to do this, given that AstExprInterpString stores the strings in a vector?

Only things I can think of are:

1. Reserving in each iteration of the `expr->strings` loop
2. Looping a second time before the big one, to count the characters
3. Pre-counting the total number of characters needed for the reserve while parsing the interpolated string

For 1 and 2, for precise counting of escaped strings you would also need to go character by character, unless you just added some acceptable fudge.

Reply:
You need to loop a second time to compute the total size, yeah. To account for `%` you can use something like this as a shorthand:

```
formatCapacity += string.size + std::count(string.data, string.data + string.size, '%');
```